include/sisl/cache/evictor.hpp

```diff
@@ -64,13 +70,14 @@ class Evictor {
 
     int64_t max_size() const { return m_max_size; }
     uint32_t num_partitions() const { return m_num_partitions; }
-    const can_evict_cb_t& can_evict_cb(const uint32_t record_id) const { return m_can_evict_cbs[record_id].second; }
+    const eviction_cb_t& can_evict_cb(const uint32_t record_id) const { return m_eviction_cbs[record_id].second.can_evict_cb; }
+    const eviction_cb_t& post_eviction_cb(const uint32_t record_id) const { return m_eviction_cbs[record_id].second.post_eviction_cb; }
 
 private:
     int64_t m_max_size;
     uint32_t m_num_partitions;
 
     std::mutex m_reg_mtx;
-    std::array< std::pair< bool, can_evict_cb_t >, CacheRecord::max_record_families() > m_can_evict_cbs;
+    std::array< std::pair< bool, RecordFamily >, CacheRecord::max_record_families() > m_eviction_cbs;
```

Comment:
nit: the `bool` here means registered/unregistered, right?
maybe `bool /*registered*/, RecordFamily`